compiler+runtime/src/cpp/jank/runtime/core.cpp

```diff
@@ -664,4 +665,48 @@ namespace jank::runtime
              expect_object<obj::inst>(o)->value.time_since_epoch())
       .count();
   }
+
+  object_ref add_watch(object_ref const reference, object_ref const key, object_ref const fn)
+  {
+    visit_object(
+      [=](auto const typed_o) -> void {
```

Comment:
We always name the visited param after the value we're visiting. So, this should be called `typed_reference`. (Same with the fn below)